core/src/smartcontracts/isi/mod.rs

```diff
@@ -614,6 +623,23 @@ impl Execute for RevokeBox {
     }
 }
 
+impl Execute for SetParameterBox {
+    type Error = Error;
+
+    fn execute(self, authority: AccountId, wsv: &WorldStateView) -> Result<(), Self::Error> {
+        let context = Context::new();
+        let (parameter, account_id) = match (
+            self.parameter.evaluate(wsv, &context)?,
+            self.source_id.evaluate(wsv, &context)?,
+        ) {
+            (Value::Parameter(parameter), IdBox::AccountId(account_id)) => (parameter, account_id),
```

Comment:
I've been fiddling back and forth between different versions of `SetParameter` and failed to notice that, thanks.

Reply:
On further examination, seems like we need to have it as a separate field if we want to introduce new permission tokens for this ISI. Otherwise it would be impossible to check for granted rights which should conform with `source_id`.